server-master/src/main/scala/com/aliyun/emr/rss/service/deploy/master/Master.scala

```diff
@@ -188,7 +188,7 @@ private[deploy] class Master(
   override def receive: PartialFunction[Any, Unit] = {
     case tm: TransportMessage if tm.getType == CHECK_FOR_WORKER_TIMEOUT =>
       executeWithLeaderChecker(null, timeoutDeadWorkers())
-    case CheckForApplicationTimeOut =>
+    case tm: TransportMessage if tm.getType == CHECK_FOR_APPLICATION_TIMEOUT =>
```

Comment:
Shall we case match TransportMessage and match getType inside? like
```
case tm: TransportMessage =>
      tm.getType match
          case CHECK_FOR_WORKER_TIMEOUT => ...
          case CHECK_FOR_APPLICATION_TIMEOUT => ...
```

Reply:
I prefer not, but acceptable if you consist, because it introduces a new nested layer,

Reply:
I think the nested one is more efficient since it only checks for TransportMessage type once, but the flat one will check for TransportMessage type each time. And eventually all messages will be of TransportMessage and we will remove the outer check then.

Reply:
make sense, updated